packages/core/src/components/mediaDeviceSelect.ts

```diff
@@ -17,42 +16,17 @@ export function setupDeviceSelector(kind: MediaDeviceKind, room?: Room) {
   const activeDeviceSubject = new BehaviorSubject<string | undefined>(undefined);
 
   const activeDeviceObservable = room
-    ? observeParticipantMedia(room.localParticipant).pipe(
-        map((participantMedia) => {
-          let localTrack: LocalAudioTrack | LocalVideoTrack | undefined;
-          switch (kind) {
-            case 'videoinput':
-              localTrack = participantMedia.cameraTrack?.track as LocalAudioTrack;
-              break;
-            case 'audioinput':
-              localTrack = participantMedia.microphoneTrack?.track as LocalVideoTrack;
-              break;
-            default:
-              localTrack = undefined;
-              break;
-          }
-          return localTrack?.mediaStreamTrack.getSettings()?.deviceId;
-        }),
-        mergeWith(activeDeviceSubject),
-      )
+    ? createActiveDeviceObservable(room, kind).pipe(mergeWith(activeDeviceSubject))
```

Comment:
do we actually need the merge with activeDeviceSubject if the room is available?